src/main.test.ts

```diff
@@ -332,88 +333,211 @@ Elapsed time:  0 ms
-- Is \`tsup.config.ts\` present, and does it conform? ❌
-  - \`tsup.config.ts\` does not exist in this project.
-- Is \`typedoc.json\` present, and does it conform? ❌
-  - \`typedoc.json\` does not exist in this project.
-- Is \`CHANGELOG.md\` present? ❌
-  - \`CHANGELOG.md\` does not exist in this project.
-- Is \`.editorconfig\` present, and does it conform? ❌
-  - \`.editorconfig\` does not exist in this project.
-- Is \`.gitattributes\` present, and does it conform? ❌
-  - \`.gitattributes\` does not exist in this project.
-- Is \`.gitignore\` present, and does it conform? ❌
-  - \`.gitignore\` does not exist in this project.
-
-Results:       0 passed, 15 failed, 15 total
+ ✘  Is the classic Yarn config file (\`.yarnrc\`) absent?
+    - The config file for Yarn Classic, \`.yarnrc\`, is present. Please upgrade this project to Yarn Modern.
+ ✘  Does the package have a well-formed manifest (\`package.json\`)?
+    - \`package.json\` does not exist in this project.
+ ✘  Is \`README.md\` present?
+    - \`README.md\` does not exist in this project.
+ ✘  Are all of the files for Yarn Modern present, and do they conform?
+    - \`.yarnrc.yml\` does not exist in this project.
+    - \`.yarn/releases/\` does not exist in this project.
+    - \`.yarn/plugins/\` does not exist in this project.
+ ✘  Does the \`src/\` directory exist?
+    - \`src/\` does not exist in this project.
+ ✘  Is \`.nvmrc\` present, and does it conform?
+    - \`.nvmrc\` does not exist in this project.
+ ✘  Is \`jest.config.js\` present, and does it conform?
+    - \`jest.config.js\` does not exist in this project.
+ ✘  Is \`tsconfig.json\` present, and does it conform?
+    - \`tsconfig.json\` does not exist in this project.
+ ✘  Is \`tsconfig.build.json\` present, and does it conform?
+    - \`tsconfig.build.json\` does not exist in this project.
+ ✘  Is \`tsup.config.ts\` present, and does it conform?
+    - \`tsup.config.ts\` does not exist in this project.
+ ✘  Is \`typedoc.json\` present, and does it conform?
+    - \`typedoc.json\` does not exist in this project.
+ ✘  Is \`CHANGELOG.md\` present?
+    - \`CHANGELOG.md\` does not exist in this project.
+ ✘  Is \`.editorconfig\` present, and does it conform?
+    - \`.editorconfig\` does not exist in this project.
+ ✘  Is \`.gitattributes\` present, and does it conform?
+    - \`.gitattributes\` does not exist in this project.
+ ✘  Is \`.gitignore\` present, and does it conform?
+    - \`.gitignore\` does not exist in this project.
+
+Results:       0 passed, 15 failed, 0 errored, 15 total
 Elapsed time:  0 ms
 
 `,
         );
+        expect(isSuccessful).toBe(false);
       });
     });
 
-    it('does not exit immediately if a project errors during linting, but shows the error and continues', async () => {
+    it('does not abort linting of a project if a lint rule throws, but shows the error and continues to the next line rule, returning false', async () => {
```

Comment:
There are two similar tests in this file which exercise the new logic added in this PR; this is one of them.